modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/import_export/ProcessImport.java

```diff
@@ -0,0 +1,219 @@
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.pc.core.import_export;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.governance.api.util.GovernanceUtils;
+import org.wso2.carbon.pc.core.ProcessCenterConstants;
+import org.wso2.carbon.pc.core.ProcessCenterException;
+import org.wso2.carbon.pc.core.audit.util.RegPermissionUtil;
+import org.wso2.carbon.pc.core.internal.ProcessCenterServerHolder;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ProcessImport {
+    private static final Log log = LogFactory.getLog(ProcessImport.class);
+    File[] listOfProcessDirs;
+    RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+    UserRegistry reg;
+    String user;
+
+    /**
+     * @param processZipInputStream
+     * @param user
+     * @throws IOException
+     * @throws RegistryException
+     * @throws ProcessCenterException
+     * @throws UserStoreException
+     */
+    public void importProcesses(InputStream processZipInputStream, String user)
+            throws IOException, RegistryException, ProcessCenterException, UserStoreException {
+
+        if (registryService != null) {
+            reg = registryService.getGovernanceUserRegistry(user);
+            this.user = user;
+            //extract zip file stream to the system disk
+            byte[] buffer = new byte[2048];
+            ZipInputStream zipInputStream = new ZipInputStream(processZipInputStream);
+            new File(ImportExportConstants.IMPORTS_DIR).mkdirs();
+
+            try {
+                ZipEntry entry;
+                while ((entry = zipInputStream.getNextEntry()) != null) {
+                    //counter++;
+                    String outpath = ImportExportConstants.IMPORTS_DIR + "/" + entry.getName();
```

Comment:
Thanks @Niranjan-K. Used Java 7 Paths and fixed with a634a28e2a8d9214670984aa763dd69483641b78